airflow/datasets/manager.py

```diff
@@ -0,0 +1,62 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from sqlalchemy.orm.session import Session
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class DatasetEventManager(LoggingMixin):
+    """
+    A pluggable class that manages operations for dataset events.
+
+    The intent is to have one place to handle all DatasetEvent-related operations, so different
+    Airflow deployments can use plugins that broadcast dataset events to each other.
+    """
+
+    def register_dataset_change(self, *, task_instance, dataset, extra=None, session: Session) -> None:
+        """
+        For local datasets, look them up, record the dataset event, queue dagruns, and broadcast
+        the dataset event
+        """
+        from airflow.datasets import Dataset
+        from airflow.models.dataset import DatasetEvent, DatasetModel
+
+        if isinstance(dataset, Dataset):
```

Comment:
not sure we need to check the type here given that the method is `register_dataset_change`

Reply:
what else would it be?  do we want pluggability to extend to "anything you can put in outlets"?

Reply:
Nope, just datasets. Fixed. :+1: